astrokat/observe_main.py

```diff
@@ -312,7 +319,11 @@ def subarray_setup(self, instrument):
             conf_param = instrument[key]
             if key == "integration_time":
                 key = "dump_rate"
-                instrument[key] = 1.0 / float(conf_param)
+                integration_time = float(conf_param)
+                if integration_time == 1.33:
+                    instrument[key] = 0.75
+                else:
+                    instrument[key] = 1.0 / integration_time
```

Comment:
The `elif conf_param != sub_sensor:` line (348), was giving the problem when comparing the dump rate due to variations in precision.  We already have some special cases based on the key, so why not just add another another for "dump_rate"?  In general, comparing floating point numbers with `==` is not a reliable approach.  Especially if those numbers are calculated in different systems.  It is quite likely that there is some tiny difference. 

```python
# Maximum difference allowed between requested and actual dump rate (Hz)
DUMP_RATE_TOLERANCE = 0.002
...

            elif key == "dump_rate":
                delta = abs(conf_param - sub_sensor)
                if delta > DUMP_RATE_TOLERANCE: 
                    raise RuntimeError(
                       "Subarray configuration {} error, {} required, "
                        "{} found, delta > tolerance ({} > {})".format(
                            sensor_name, conf_param, sub_sensor, delta, DUMP_RATE_TOLERANCE)
                    )
            elif conf_param != sub_sensor:
                raise RuntimeError(
                    "Subarray configuration {} error, {} required, "
                    "{} found".format(sensor_name, conf_param, sub_sensor)
                )
```

I'd prefer a tolerance of 0.001 Hz, but then the OPT needs to use an integration time of 1.333 seconds instead of 1.33.  Is that possible?

Reply:
done